### DISCUSSION TITLE
Am I flashing to the NVMe drive correctly?

### DISCUSSION THREAD

**Discussion by gagestinson** [Joshua-Riek/ubuntu-rockchip#144] [Q&A]
Hello,

I am fairly new to Linux and am trying to set up a home networking system. I got the Orange Pi 5 since the Raspberry Pi 4 is unavailable. Plus the Orange Pi 5 has better specs anyway.

I am trying to get normal Ubuntu onto it instead of the Orange Pi 5 Ubuntu version on the Orange Pi website. I will list how I am trying to set this up below.

On my personal laptop, I use BelenaEtcher to flash this image onto a SD card. I then boot the Orange Pi from the SD card. I run the `sudo dd if=/lib/u-boot-orangepi-rk3588/rkspi_loader.img of=/dev/mtdblock0 conv=notrunc` command as it says to on the wiki. I then reboot. But upon reboot, the NVMe SSD I have is not available in the file system. It is shown when running `sudo fdisk -l` but it is not in the file explorer.

I have cleared the partitions on the mtdblock0 and on the NVMe. I have also tried incorporating this [tutorial](https://www.crosstalksolutions.com/orange-pi-5-simple-overview-and-installation-with-m-2-ssd/) by flashing it to the mtblock0 as well as to the NVMe by simply altering the command given in that tutorial to `sudo dd bs=1M if=ubuntu-22.04.2-preinstalled-desktop-arm64-orangepi5.img.xz of=/dev/nvme0n1 status=progress`

Any advice would be greatly appreciated.

**Comment by Joshua-Riek** [comment]
The os image is compressed, so you must uncompress it and pipe it to the dd utility like below.
```
xz -dc ubuntu-22.04.2-preinstalled-desktop-arm64-orangepi5.img.xz | sudo dd of=/dev/nvme0n1 bs=4k
```
I want to add Etcher to the os image by default at some point to make this easier for users, but I have not had the time to do so yet.
